lib/hiera/backend.rb

```diff
@@ -196,14 +196,30 @@ def merge_answer(left,right)
       # Backend instances are cached so if you need to connect to any
       # databases then do so in your constructor, future calls to your
       # backend will not create new instances
+
+      # @param key [String] The key use when doing the lookup
+      # @param scope [#[]] The primary source of data for substitutions.
+      # @param order_override [String] An override that will be pre-pended to the hierarchy definition. Can be nil
+      # @param resolution_type [Symbol] One of :hash, :array, or :priority. Can be nil which is the same as :priority
+      # return The value that corresponds to the given key or nil if no such value cannot be found
+      #
       def lookup(key, default, scope, order_override, resolution_type)
         @backends ||= {}
         answer = nil
 
+        segments = key.split('.')
+        backend_resolution_type = resolution_type
+        subsegments = nil
+        if segments.size > 1
+          backend_resolution_type = :hash
```

Comment:
Is that really wanted here? IIRC this will do a hash merge across all layers as opposed to returning a hash bound to a key.